### PR TITLE
[READY] Improvised Makeshift Pistol

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

This pull requests adds a new firearm in the default crafting menu. This pistol is chambered in standard 9mm bullets and can hold up to 5 bullets in its magazine. The pistol fires at a delay of 1 second and uses the standard bolt you can find on the Makarov pistol. To craft it, you need 2 Plasteel sheets, 5 Metal sheets, 1 Modular Receiver and a pipe. It comes with a magazine but you can craft more using 2 Plasteel Sheets and 2 Metal Sheets for a magazine. Whether or not you will be able to craft magazines in the future is to be seen as this may cause a balance issue.

This has been tested and the pistol is working.

## Why It's Good For The Game

Previous to this PR, there were 2 ranged firearms that you could craft by default. The Flamethrower and the Makeshift Shotgun. While both of these weapons fill their niches perfectly (Flamethrower for crowd control/short range and shotgun for short range), there are no weapons to be used for medium to long range combat. While yes, you can outfit the shotgun with a slug and use it for medium/long range, the fact that you only have a single shot before you need to reload puts you at a massive disadvantage since if you miss you will need to manually reload before being able to shoot again. With the pistol, if you miss you can simply lead and shoot again for another 4 times before needing a reload. You are still at a disadvantage to people using proper weapons however since they will most likely have higher capacity weapons with higher bullet velocities than the pistol's.

This weapon should be treated as CONTRABAND by any Security Officer that sees someone holding it, since it can do quite a bit of damage if used in the right hands. Arrest is warranted if seen with one.

TL;DR 
This is good for the game because it fills a niche the other craftable weapons cannot.

Below are some images of the pistol. 

image

Note that the below image is an earlier sprite of the gun. The off-centered sprite has also been fixed.
image

PS: This was originally going to be a Fulp exclusive as that is where I first sent this PR but I wanted all the downstreams to benefit from it so I sent it here.

## Changelog
:cl:
add: Added pistol
add: Added better sprite for pistol
add: Changed the ammo type from 10mm to 9mm since it was too damaging.
add: Added insuls to the recipe, making it more difficult to create the weapon.

/:cl:

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
>fact that you only have a single shot before you need to reload puts you at a massive disadvantage

but thats the point? makeshift weapons are supposed to have a downside to outweigh the fact that you can just print them out for free

also why'd you model it after the liberator? that was one of the only modern single shot 
pistols ever mass produced

File: code/modules/projectiles/boxes_magazines/external/pistol.dm [hunk 1]
```diff
@@ -83,3 +83,13 @@
 	caliber = ".50"
 	max_ammo = 7
 	multiple_sprites = AMMO_BOX_PER_BULLET
+
+/obj/item/ammo_box/magazine/liberator10mm
```
Comment (MrDoomBringer) [inline code/modules/projectiles/boxes_magazines/external/pistol.dm:87]: you know this will pre-load it with bullets right?

File: code/modules/projectiles/guns/ballistic/pistol.dm [hunk 2]
```diff
@@ -97,3 +97,17 @@
 		to_chat(user, "<span class='notice'>..and falls into view. Whew, that was a close one.</span>")
 		user.dropItemToGround(src)
 
+/obj/item/gun/ballistic/automatic/liberatorpistol
+	name = "Makeshift Pistol"
+	desc = "Harkening back to the prehistoric Liberator pistol design, this small makeshift pistol fires 10mm bullets."
```
Comment (fludd12) [inline code/modules/projectiles/guns/ballistic/pistol.dm:102]:
```suggestion
	desc = "Harkening back to before humanity achieved spaceflight on Terra, this \"Liberator\" makeshift pistol fires 10mm bullets."
```
Gives it a bit more of a spacebound-humanity-looking-back-on-earthbound-humanity feel. Also doesn't imply that we lost most of Earth's history of the mid-20th century.

File: code/modules/projectiles/boxes_magazines/external/pistol.dm [hunk 3]
```diff
@@ -83,3 +83,13 @@
 	caliber = ".50"
 	max_ammo = 7
 	multiple_sprites = AMMO_BOX_PER_BULLET
+
+/obj/item/ammo_box/magazine/liberator10mm
+	name = "Makeshift 9mm magazine"
+	desc = "A Makeshift Pistol magazine. Load with 9mm bullets. Usable in Makeshift Pistol."
+	icon_state = "libmag"
+	ammo_type = /obj/item/ammo_casing/c9mm
```
Comment (MrDoomBringer) [inline code/modules/projectiles/boxes_magazines/external/pistol.dm:91]: are you sure you want people to be doing 30 damage? You could kill someone in less than 2 seconds from across the screen if you just made two of these

File: code/modules/projectiles/guns/ballistic/pistol.dm [hunk 4]
```diff
@@ -97,3 +97,17 @@
...
+	icon_state = "liberatorpistol"
+	item_state = "liberatorpistol"
```
Comment (MrDoomBringer) [inline code/modules/projectiles/guns/ballistic/pistol.dm:104]: can you not call it liberatorpistol, give it a name that makes sense to a reader who might be looking through DMI files

File: code/modules/projectiles/boxes_magazines/external/pistol.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/modules/projectiles/boxes_magazines/external/pistol.dm:87]: I would tell you to set var/start_empty = TRUE but I have no idea where you expect people to be getting their 9mm bullets from

File: code/modules/projectiles/boxes_magazines/external/pistol.dm [hunk 5]
```diff
@@ -83,3 +83,13 @@
 	caliber = ".50"
 	max_ammo = 7
 	multiple_sprites = AMMO_BOX_PER_BULLET
+
+/obj/item/ammo_box/magazine/liberator10mm
+	name = "Makeshift 9mm magazine"
+	desc = "A Makeshift Pistol magazine. Load with 9mm bullets. Usable in Makeshift Pistol."
```
Comment (fludd12) [inline code/modules/projectiles/boxes_magazines/external/pistol.dm:89]:
```suggestion
	desc = "A handcrafted ammunition magazine, designed to be loaded with 10mm bullets. It seems to be custom-fit for an off-market pistol design."
```
The description of the pistol, and the path name, implies that this is 10mm, not 9mm. Also, we needn't call out the makeshift pistol by name here -- it's not like random spessmen have the time to be writing instructions for use on their illicit handmade firearm equipment.

**Comment by PapaSejd** [comment]
I modeled it after the Liberator simply because it looks almost handmade (and was described to be borderline that) and it would fit what this would most likely be used for (Tators hiding in maint LARPing guerilla warfare with it, or maybe just an assistant doing the same). I differed from the Liberator with the multi shot capability was to give us a projectile-based weapon that was able to shoot more than once since the only weapon other than this that does that is the flamethrower, which as I said was only good for short range. Ill admit that the recipe is still too cheap for a weapon that can shoot more than once so I might add a completely new component for bolt that uses rarer parts from science.

File: code/modules/projectiles/boxes_magazines/external/pistol.dm (hunk 5 above)
Comment (PapaSejd) [reply code/modules/projectiles/boxes_magazines/external/pistol.dm:89]: Thanks for fixing this. I wrote the descriptions yesterday when this weapon was still 10mm and I changed it to 9mm this morning forgetting that I had to change the desc too.

File: code/modules/projectiles/guns/ballistic/pistol.dm [hunk 6]
```diff
@@ -97,3 +97,17 @@
 		to_chat(user, "<span class='notice'>..and falls into view. Whew, that was a close one.</span>")
 		user.dropItemToGround(src)
 
+/obj/item/gun/ballistic/automatic/liberatorpistol
```
Comment (MrDoomBringer) [inline code/modules/projectiles/guns/ballistic/pistol.dm:100]: Why did you make this an automatic gun then set the bust_size to 1?? That defeats the entire point, plus people will be able to switch firemodes to remove the fire_delay effect. Just make this a normal ballistic subtype, unless i'm missing something

File: code/modules/projectiles/guns/ballistic/pistol.dm [hunk 7]
```diff
@@ -97,3 +97,17 @@
...
+	name = "Makeshift Pistol"
+	desc = "Harkening back to the prehistoric Liberator pistol design, this small makeshift pistol fires 10mm bullets."
+	icon_state = "liberatorpistol"
+	item_state = "liberatorpistol"
+	w_class = WEIGHT_CLASS_SMALL
+	mag_type = /obj/item/ammo_box/magazine/liberator10mm
```
Comment (MrDoomBringer) [inline code/modules/projectiles/guns/ballistic/pistol.dm:106]:
```suggestion
	mag_type = /obj/item/ammo_box/magazine/liberator9mm
```

File: code/modules/projectiles/boxes_magazines/external/pistol.dm (hunk 1 above)
Comment (MrDoomBringer) [inline code/modules/projectiles/boxes_magazines/external/pistol.dm:87]:
```suggestion
/obj/item/ammo_box/magazine/liberator9mm

```

File: code/datums/components/crafting/recipes.dm [hunk 8]
```diff
@@ -835,3 +835,25 @@
...
+				/obj/item/stack/sheet/plasteel = 2,
+				/obj/item/stack/sheet/metal = 5)
+	tools = list(TOOL_WELDER)
+	time = 100
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
+
+/datum/crafting_recipe/makeshiftliberatorpistolmagazine
+	name = "Improvised Pistol Magazine"
+	result = /obj/item/ammo_box/magazine/liberator10mm
```
Comment (MrDoomBringer) [inline code/datums/components/crafting/recipes.dm:853]:
```suggestion
	result = /obj/item/ammo_box/magazine/liberator9mm
```

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]
uuuh you should take a better look at some of the balance issues with this

ic its ur first PR tho so all in all not bad!

**Comment by MrDoomBringer** [comment]
image

make it look like an eoka or something tbpqh

File: code/modules/projectiles/boxes_magazines/external/pistol.dm (hunk 1 above)
Comment (PapaSejd) [reply code/modules/projectiles/boxes_magazines/external/pistol.dm:87]: Most likely from the hacked autolathe. If they dont have access to this, they get the 5 bullets so that the weapon isnt completely useless upon creation.

File: code/modules/projectiles/guns/ballistic/pistol.dm (hunk 6 above)
Comment (PapaSejd) [reply code/modules/projectiles/guns/ballistic/pistol.dm:100]: Wait I think that was a mistake by me, lemme check this

**Comment by fludd12** [comment]
My suggestion, if you're looking to make it harder to craft, is to simply use more varied tools and materials that can still be scavenged from the station, rather than requiring a trip to science. Having parts be science-bound tends to make the weapons really only made by science in their spare time.

You could, for example, make the gun only able to be crafted with metal, plasteel, and wood at a table with certain specific tools. Or make parts of the gun require items that the average player won't want to give up so easily, such as requiring a rubber sealant that can only be obtained by using insulated gloves in the recipe and having wirecutters as a tool.

Note that it doesn't always have to make perfect sense, just as long as a player can internally justify why they needed the item based on the information you give. If the description for the gun says it required rugged rubber to seal the firing chamber and provide a cushion so it doesn't fly apart, for example, most players would be able to suspend disbelief and say "oh, yeah, that's why I had to tear up my insulated gloves for it." That's, of course, just an example. It could just as easily require a cake hat for its' intricate ignition mechanism or some other bullshit. The world is your oyster!

**Comment by PapaSejd** [comment]
image
I mean this looks like you took random pieces of metal, welded it together and called it a day. I wouldnt trust it if I had to fire it.

**Comment by PapaSejd** [comment]
> My suggestion, if you're looking to make it harder to craft, is to simply use more varied tools and materials that can still be scavenged from the station, rather than requiring a trip to science. Having parts be science-bound tends to make the weapons really only made by science in their spare time.
> 
> You could, for example, make the gun only able to be crafted with metal, plasteel, and wood at a table with certain specific tools. Or make parts of the gun require items that the average player won't want to give up so easily, such as requiring a rubber sealant that can only be obtained by using insulated gloves in the recipe and having wirecutters as a tool.

Ill check in the objects folder for materials that would suit the weapon and add some difficulty.

**Comment by PapaSejd** [comment]
Actually the insul idea is pretty great, maybe the barrel needs to to _insulate_ it so that it doesnt heat up the rest of the gun (since its all metal).

File: code/modules/projectiles/boxes_magazines/external/pistol.dm (hunk 3 above)
Comment (PapaSejd) [reply code/modules/projectiles/boxes_magazines/external/pistol.dm:91]: True, but at those distances your target is going to have to not be whizzing around since bullets in SS13 arent exactly faster than sound

**Comment by PapaSejd** [comment]
Now that insuls are in the recipe, only those who rush the Tool Shed or have access to the engineering insuls can create the weapon, which is a deal harder than the Improv shotty that only needs a pipe, wood, receiver and some packaging.

**Comment by ExcessiveUseOfCobblestone** [comment]
I think we made a mistake with makeshift shotguns given how they're used ingame, I don't believe limited ammo capacity in a game where you can fit these bullets within boxes and these can be accessed just as easily is an adequate tradeoff for the ability to make these.

Makeshift items should have a chance to backfire because they are not properly made, scaling with use, rather than just be less powerful than their counterparts.

Personally if I had the inclination to make a makeshift weapon, I'd just make a shotgun.

**Comment by PapaSejd** [comment]
@ExcessiveUseOfCobblestone 

Ill check to see if I can give the gun a chance to malfunction (maybe 1/3 chance?) and even explode (1/10 - 1/20?) when used. However if this is added ill have to scale back on the recipe as its incredibly expensive at the moment. 

Jamming and exploding would probably warrant a drop in price for the recipe to:

- 5 Sheets of metal
- Pipe 
- Modular Receiver
- 2 Rods

**Comment by Archanial** [comment]
Don't make it explode, noone will use it.
Keep the recipe as it is so people won't make lot's of them and just avoid jamming by switching guns.

**Comment by PapaSejd** [comment]
@Archanial Got it. Ill keep it as it is. The insuls should keep powergaming assistants for getting the gun immediately but it shouldn't make it absurdly hard to get. Other than that I believe the gun is ready to be merged.

**Comment by Archanial** [comment]
Also 1/3 seems like too high of a chance to jam. 1/5-1/7 should be good? Or make it have higher chance of jamming with each fired shot.

**Comment by PapaSejd** [comment]
Personally I dont think the gun will require jamming to be balanced since weapons like the improv shotgun and flamethrower never jam yet ive never seen them be largely abused. Both are incredibly cheap to make (the flamethrower is printable in a hacked autolathe) but are rarely abused by powergamers.

**Comment by optimumtact** [comment]
Things we don't need
more improvised weapons

File: code/modules/projectiles/boxes_magazines/external/pistol.dm (hunk 3 above)
Comment (ThePainkiller) [reply code/modules/projectiles/boxes_magazines/external/pistol.dm:91]: My suggestion is that since it's a ghetto pistol make it either very inacurate with high spread, or have a chance of backfire like a tinkered det's revolver.

**Comment by Archanial** [comment]
> 
> 
> Things we don't need
> more improvised weapons

why?

**Comment by MrDoomBringer** [comment]
we have enough improvised weapons as is
people should be able to kill people with guns through creative use of the station and the tools around them.
encouraging people to just make their own slightly shittier weapons and deathmatch the people with the better weapons is a little dumb
its a huge hassle to balance makeshift weapons properly

uuh i dunno those are the reasons I could think up over the course of like 30 seconds, i probably don't need to make mr orange man think of many more for me
